packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -235,4 +235,58 @@ export class Span implements api.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  // Utility function to truncate given value within size
+  // for value type of string, will truncate to given limit
+  // for type of non-string, will return same value
+  private _truncateToLimitUtil(value: string, limit: number): string {
+    if (value.length <= limit) {
+      return value;
+    }
+    return value.substr(0, limit);
+  }
+
+  // Check whether given value is array of strings or not
+  private _isArrayOfStrings(value: unknown): boolean {
+    if (!Array.isArray(value)) {
+      return false;
+    }
+    return value.every(val => typeof val == 'string');
+  }
+
+  /**
+   * If the given attribute value is of type string and has more characters than given {@code attributeValueLengthLimit} then
+   * return string with trucated to {@code attributeValueLengthLimit} characters
+   *
+   * If the given attribute value is array of strings then
+   * return new array of strings with each element truncated to {@code attributeValueLengthLimit} characters
+   *
+   * Otherwise return same Attribute {@code value}
+   *
+   * @param value Attribute value
+   * @returns truncated attribute value if required, otherwise same value
+   */
+  private _truncateToSize(value?: SpanAttributeValue): SpanAttributeValue | undefined {
+    const limit = this._spanLimits.attributeValueLengthLimit;
+    // Check limit
+    // undefined limit means do not truncate
+    if (typeof limit != 'number' || limit <= 0) {
+      // Non-integer and negative values are invalid, so do not truncate
+      api.diag.warn(`Attribute value limit must be positive, got ${limit}`);
```

Comment:
We are warning when the limit is undefined?

I would prefer to have an explicit shortcut at the top like

```typescript
if (limit == null || value == null) {
  return value
}
```

Reply:
I tried this shortcut, but not able to cover this test case.
since

for `value == null` wont come to this function since parent function `setAttribute` is handled it at line 91.

and default value for `limit=Infinity` so `limit == null` also won't execute all the times.

we are making check `typeof limit != 'number'` to avoid limit of `undefined` type